engineapi/engineapi/middleware.py

```diff
@@ -199,3 +212,157 @@ async def dispatch(
         request.state.token = user_token
 
         return await call_next(request)
+
+
+def parse_origins_from_resources(origins: List[str]) -> Set[str]:
```

Comment:
Does the resources here refer to Brood resources?

Reply:
Yes